turbo/engineapi/engine_server.go

```diff
@@ -751,7 +751,7 @@ func (e *EngineServer) HandleNewPayload(
 			return &engine_types.PayloadStatus{Status: engine_types.SyncingStatus}, nil
 		}
 
-		if !e.blockDownloader.StartDownloading(ctx, 0, header.ParentHash, block) {
+		if !e.blockDownloader.StartDownloading(ctx, 0, header.Hash(), block) {
```

Comment:
This change doesn't look right. We're missing the parent of a new payload (`block`) coming from CL, so we need to only download the parent.

Reply:
Ok, I switched to starting the download from the parent. I had some issues if the latest block itself wasn't downloaded when restarting the node and re-triggering the download, but it could be due to something else also. So I am switching it back to how it was now, and just setting the correct block height.